### PR TITLE
BLAC-9. Sorting men's shoes(full test).

### PR DESCRIPTION
**Pull request by DashaYab**

### CODE REVIEW

File: src/test/java/elements/HomeElements.java [hunk 1]
```diff
@@ -14,21 +14,21 @@ public HomeElements(WebDriver driver) {
...
-    public static final String LOGIN_BUTTON_ON_ACCOUNT_MENU = "//a[@data-test-id='login-button']";
-    public static final String REGISTER_HERE_BUTTON = "//a[@data-test-id='register-button']";
+    public static final String LOGIN_BUTTON_ON_ACCOUNT_MENU = dataTestId("login-button");
+    public static final String REGISTER_HERE_BUTTON = dataTestId("register-button");
     public static final String COLLABORATION_BTN = "//span[contains(text(),'Collaborations')]";
-    public static final String SEARCH_BTN ="//button[@type='submit']//div";
+    public static final String SEARCH_BTN = "//button[@type='submit']//div";
     public static final String SEARCH_INPUT_FIELD = "//input[@placeholder='SEARCH PUMA.COM']";
-    public static final String SEARCH_FIELD = "//button[@data-test-id='search-icon-nav']";
+    public static final String SEARCH_FIELD = dataTestId("search-icon-nav");
```
Comment (AndriiMyronenko91) [inline src/test/java/elements/HomeElements.java]: please use incapsulation OOP principle

File: src/test/java/elements/HomeElements.java [hunk 2]
```diff
@@ -38,8 +38,8 @@ public HomeElements(WebDriver driver) {
     public static final String PRICE_LOW_TO_HIGH_SORTING = "//select[@data-test-id='product-list-sort-select-input']//option[@value='price-low-to-high']";
     public static final String PRICE_HIGH_TO_LOW_SORTING = "//select[@data-test-id='product-list-sort-select-input']//option[@value='price-high-to-low']";
     public static final String NEWEST_PRICE_SORTING = "//select[@data-test-id='product-list-sort-select-input']//option[@value='newest']";
-    public static final String MEN_SOCCER_SHOES_PRICE = "//span[@data-test-id='price']";
     public static final String LIST_OF_PRICE = "//div[@class='relative flex mobile:flex-col w-full items-start']/div/span[1]";
+    public static final String LIST_OF_NEWEST_PRODUCT = dataTestId("product-badge");
```
Comment (AndriiMyronenko91) [inline src/test/java/elements/HomeElements.java:46]: OOP incapsulation

File: src/test/java/elements/ProductCartElements.java [hunk 3]
```diff
@@ -8,12 +8,13 @@ public class ProductCartElements extends AbstractBasePage {
...
-    public WebElement getAddToCartBtn() {
-        return waitUntilElementToBeClickableByXpath(ADD_TO_CART_BTN);
-    }
+    
     public static final String SIZE_VALUE = "(//span[@data-content='size-value'])[1]";
-    public static final String ADD_TO_CART_BTN = "//button[@data-test-id='pdp-add-to-cart']";
-    public static final String VIEW_CART_BTN = "//a[@data-test-id='minicart-cart-link']";
+    public static final String ADD_TO_CART_BTN = dataTestId("pdp-add-to-cart");
+    public static final String VIEW_CART_BTN = dataTestId("minicart-cart-link");
     public static final String SIZE_FIELD = "(//label[@data-disabled='false'])[1]//span[@data-content='size-value']";
```
Comment (AndriiMyronenko91) [inline src/test/java/elements/ProductCartElements.java:15]: ditto

File: src/test/java/elements/RegisterElements.java [hunk 4]
```diff
@@ -10,21 +10,21 @@ public RegisterElements(WebDriver driver) {
...
-    public static final String REGISTER_BUTTON = "//button[@data-test-id='register']";
-    public static final String LOGIN_ALREADY_IN_USE_TEXT = "//p[@data-test-id='registration-form-error-content']";
-    public static final String CREATE_ACCOUNT_BUTTON_ON_TOP = "//a[@data-test-id='register-action-button']";
+    public static final String FIRST_NAME_FIELD = dataTestId("first-name");
+    public static final String LAST_NAME_FIELD = dataTestId("last-name");
+    public static final String EMAIL_FIELD = dataTestId("email");
+    public static final String PASSWORD_FIELD = dataTestId("password");
+    public static final String REGISTER_BUTTON = dataTestId("register");
+    public static final String LOGIN_ALREADY_IN_USE_TEXT = dataTestId("registration-form-error-content");
+    public static final String CREATE_ACCOUNT_BUTTON_ON_TOP = dataTestId("register-action-button");
```
Comment (AndriiMyronenko91) [inline src/test/java/elements/RegisterElements.java:19]: ditto

File: src/test/java/elements/RegisterElements.java [hunk 5]
```diff
@@ -10,21 +10,21 @@ public RegisterElements(WebDriver driver) {
...
-    public static final String EMAIL_FIELD_IN_FORGOTTEN_PASSWORD = "//input[@data-test-id='email-input']";
-    public static final String EMAIL_FIELD_ON_LOGIN_PAGE = "//input[@data-test-id='auth-field-email']";
-    public static final String PASSWORD_FIELD_ON_LOGIN_PAGE = "//input[@data-test-id='auth-field-password']";
-    public static final String LOGIN_BUTTON_ON_LOGIN_PAGE = "//button[@data-test-id='auth-button-login']";
+    public static final String FORGOTTEN_YOUR_PASSWORD = dataTestId("forgotten-password-link");
+    public static final String SUBMIT_IN_FORGOTTEN_PASSWORD = dataTestId("submit-password-reset-btn");
+    public static final String EMAIL_FIELD_IN_FORGOTTEN_PASSWORD = dataTestId("email-input");
+    public static final String EMAIL_FIELD_ON_LOGIN_PAGE = dataTestId("auth-field-email");
+    public static final String PASSWORD_FIELD_ON_LOGIN_PAGE = dataTestId("auth-field-password");
+    public static final String LOGIN_BUTTON_ON_LOGIN_PAGE = dataTestId("auth-button-login");
```
Comment (AndriiMyronenko91) [inline src/test/java/elements/RegisterElements.java:27]: ditto

File: src/test/java/elements/SelectProductElements.java [hunk 6]
```diff
@@ -8,6 +8,6 @@ public SelectProductElements(WebDriver driver) {
         super(driver);
     }
 
-    public static final String PRODUCT_TITTLE = "//h1[@data-test-id='pdp-title']";
-    public static final String PRODUCT_PRICE = "//span[@data-test-id='item-price-pdp']";
+    public static final String PRODUCT_TITTLE = dataTestId("pdp-title");
+    public static final String PRODUCT_PRICE ="//div[@class='flex flex-col pb-1']/span[1]";
```
Comment (AndriiMyronenko91) [inline src/test/java/elements/SelectProductElements.java:12]: ditto

Comment (AndriiMyronenko91) [inline src/test/java/elements/SelectProductElements.java:12]: write better Xpath, please avoid [1] and such numbers in xpath

File: src/test/java/pages/ProductCardPage.java [hunk 7]
```diff
@@ -3,24 +3,27 @@
 import elements.ProductCartElements;
 import org.openqa.selenium.WebDriver;
 
+import static base.CommonActions.scrollByPixel;
+
 public class ProductCardPage extends ProductCartElements {
```
Comment (AndriiMyronenko91) [inline src/test/java/pages/ProductCardPage.java:8]: as i remember we add empty lines after class

File: src/test/java/tests/ShoppingCartTest.java [hunk 8]
```diff
@@ -68,9 +68,8 @@ public void checkShoppingCart() throws InterruptedException {
 
         shoppingCartPage
                 .getCartProductTittle("2")
-                .clickValueBtn(1)
                 .selectValue("2");
-        sleep(5000);
+        sleep(3000);
```
Comment (AndriiMyronenko91) [inline src/test/java/tests/ShoppingCartTest.java:72]: why do you need sleep here?

**Review by AndriiMyronenko91 — CHANGES_REQUESTED** [review]

File: src/test/java/tests/ShoppingCartTest.java (hunk 8 above)
Comment (DashaYab) [reply src/test/java/tests/ShoppingCartTest.java:72]: Doesn't have time to change the price on the site when the selected product is increased... without sleep test failed.I tried to take the sleep to another place(on the "selectValue"), but the test passed every other time.Therefore, I reduced the duration of sleep to the minimum at which the test passes...

Comment (AndriiMyronenko91) [reply src/test/java/tests/ShoppingCartTest.java:72]: Can you use some kind of waiter instead?

File: src/test/java/elements/SelectProductElements.java (hunk 6 above)
Comment (DashaYab) [reply src/test/java/elements/SelectProductElements.java:12]: I tried to make a better Хpath, but it didn't succeed. Maybe you can help me?

Comment (AndriiMyronenko91) [reply src/test/java/elements/SelectProductElements.java:12]: Try to copy part of the DOM to chatGPT and ask it to do using axes

File: src/test/java/tests/ShoppingCartTest.java (hunk 8 above)
Comment (DashaYab) [reply src/test/java/tests/ShoppingCartTest.java:72]: I use waiter(on the "selectValue") but it's not enough

File: src/test/java/elements/SelectProductElements.java (hunk 6 above)
Comment (DashaYab) [reply src/test/java/elements/SelectProductElements.java:12]: Thank you)

**Review by Max2306M — APPROVED** [review]

**Review by AndriiMyronenko91 — APPROVED** [review]

**Review by AndriiMyronenko91 — APPROVED** [review]

**Review by Obibus — APPROVED** [review]

**Pull request merged**